packages/xrpl/test/integration/transactions/permissionedDEX.test.ts

```diff
@@ -0,0 +1,373 @@
+import { stringToHex } from '@xrplf/isomorphic/utils'
+import { assert } from 'chai'
+
+import {
+  PermissionedDomainSet,
+  AuthorizeCredential,
+  Wallet,
+  IssuedCurrencyAmount,
+  AccountSetAsfFlags,
+} from '../../../src'
+import DirectoryNode from '../../../src/models/ledger/DirectoryNode'
+import Offer from '../../../src/models/ledger/Offer'
+import PermissionedDomain from '../../../src/models/ledger/PermissionedDomain'
+import {
+  BookOffersRequest,
+  BookOffersResponse,
+} from '../../../src/models/methods/bookOffers'
+import { SubmitResponse } from '../../../src/models/methods/submit'
+import {
+  SubscribeBook,
+  SubscribeRequest,
+} from '../../../src/models/methods/subscribe'
+import { CredentialAccept } from '../../../src/models/transactions/CredentialAccept'
+import { CredentialCreate } from '../../../src/models/transactions/CredentialCreate'
+import {
+  OfferCreate,
+  OfferCreateFlags,
+} from '../../../src/models/transactions/offerCreate'
+import { Payment } from '../../../src/models/transactions/payment'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { generateFundedWallet, testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('PermissionedDEX', function () {
+  let testContext: XrplIntegrationTestContext
+  let wallet1: Wallet
+  let wallet2: Wallet
+  let pd_ledger_object: PermissionedDomain
+  let offerCreateTxResponse: SubmitResponse
+  let offer_ledger_object: Offer
+
+  beforeAll(async () => {
+    // this section describes the pre-requisites for testing PermissionedDEX features
+    testContext = await setupClient(serverUrl)
+    wallet1 = await generateFundedWallet(testContext.client)
+    wallet2 = await generateFundedWallet(testContext.client)
```

Comment:
Can you rename these wallets to their roles in this test?

Reply:
Hello Omar, 
I'm unable to identify `roles` for wallet1 and wallet2. Both of them participate in the PermissionedDEX orderbook. Both the wallets create permissioned-offers which cross each other.

Both of them are equivalent. Unlike the example of an escrow, there is no `creator` / `destination` in the case of Permissioned Offers.